crates/network-tunnel/src/sshforwarding.rs

```diff
@@ -78,210 +69,89 @@ fn port_schema(title: &str, description: &str) -> schemars::schema::Schema {
-        )
-        .map(|key| key::KeyPair::RSA {
-            key,
-            hash: key::SignatureHash::SHA2_256,
-        });
-
-        let openssh_key_pair = openssh::decode_openssh(
-            &Self::read_openssh_key(self.config.private_key.clone())?,
-            None,
-        );
-
-        let key_pair = Arc::new(rsa_key_pair.or(openssh_key_pair)?);
-
-        let sc = self
-            .ssh_client
-            .as_mut()
-            .expect("ssh_client is uninitialized.");
-        if !sc
-            .authenticate_publickey(&self.config.user, key_pair)
-            .await?
-        {
-            return Err(Error::InvalidSshCredential);
+            process: None,
         }
-
-        Ok(())
     }
 }
 
 #[async_trait]
 impl NetworkTunnel for SshForwarding {
     async fn prepare(&mut self) -> Result<(), Error> {
-        self.prepare_ssh_client().await?;
-        self.prepare_local_listener().await?;
-        self.authenticate().await?;
-        Ok(())
-    }
-
-    async fn start_serve(&mut self) -> Result<(), Error> {
-        let sc = self
-            .ssh_client
-            .as_mut()
-            .expect("ssh_client is uninitialized.");
-        let ll = self
-            .local_listener
-            .as_mut()
-            .expect("local_listener is uninitialized.");
+        // Write the key to a temporary file
+        let mut temp_key_path = std::env::temp_dir();
+        temp_key_path.push("id_rsa");
+
+        tokio::fs::write(&temp_key_path, self.config.private_key.as_bytes()).await?;
+        tokio::fs::set_permissions(&temp_key_path, std::fs::Permissions::from_mode(0o600)).await?;
+
+        let local_port = self.config.local_port;
+        let forward_host = &self.config.forward_host;
+        let forward_port = self.config.forward_port;
+
+        let mut child = Command::new("ssh")
+            .args(vec![
+                // Disable psuedo-terminal allocation
+                "-T".to_string(),
+                // Be verbose so we can pick up signals about status of the tunnel
+                "-v".to_string(),
+                // This is necessary unless we also ask for the public key from users
+                "-o".to_string(),
+                "StrictHostKeyChecking no".to_string(),
+                // Pass the private key
+                "-i".to_string(),
+                temp_key_path.into_os_string().into_string().unwrap(),
+                // Do not execute a remote command. Just forward the ports.
+                "-N".to_string(),
+                // Port forwarding stanza
+                "-L".to_string(),
+                format!("{local_port}:{forward_host}:{forward_port}"),
+                self.config.ssh_endpoint.clone(),
+            ])
+            .stderr(Stdio::piped())
+            .spawn()?;
+
+        // Read stderr of SSH until we find "Local forwarding listening", which means
+        // the ports are open and we are ready to serve requests
```

Comment:
Let me know if there is a better way of doing this... I'll also do a bit of search to see if I can find a better solution